chapter_attention-mechanisms/bahdanau-attention.md

```diff
@@ -310,12 +371,45 @@ for eng, fra in zip(engs, fras):
 ```
 
 ```{.python .input}
-#@tab all
+#@tab pytorch
+engs = ['go .', "i lost .", 'he\'s calm .', 'i\'m home .']
+fras = ['va !', 'j\'ai perdu .', 'il est calme .', 'je suis chez moi .']
+for eng, fra in zip(engs, fras):
+    translation, dec_attention_weight_seq = d2l.predict_seq2seq(
+        net, eng, src_vocab, tgt_vocab, num_steps, device, True)
+    print(f'{eng} => {translation}, ',
+          f'bleu {d2l.bleu(translation, fra, k=2):.3f}')
+```
+
+```{.python .input}
+#@tab tensorflow
+engs = ['go .', "i lost .", 'he\'s calm .', 'i\'m home .']
+fras = ['va !', 'j\'ai perdu .', 'il est calme .', 'je suis chez moi .']
+for eng, fra in zip(engs, fras):
+    translation, dec_attention_weight_seq = d2l.predict_seq2seq(net, eng, src_vocab, tgt_vocab, num_steps, True)
+    print(f'{eng} => {translation}, ',
+          f'bleu {d2l.bleu(fra, translation, k=2):.3f}')
```

Comment:
For consistency to allow for shared code blocks: why do we have to switch the order of `fra` and `translation`? Why is `device` missing in `d2l.predict_seq2seq`?

Reply:
This is an unintentional error. I think I should explain why it happened at the first place. In TF, the convention of using loss function is `loss_function(true_values, predictions)`. So in the first version of the code, I had followed TF convention and written `bleu` to take `ture_values` as first argument and `predictions` as second argument. And I had followed that in other sections as well. But while creating a pull request, I thought of following D2L's convention and changed `bleu`. However, unfortunately I forgot to change it at few other places that were promptly detected by you. But thanks to @terrytangyuan , it's fixed now in [1782](https://github.com/d2l-ai/d2l-en/pull/1782).